process/block/preprocess/intermediateResults.go

```diff
@@ -50,16 +50,17 @@ func NewIntermediateResultsProcessor(
 		adrConv:          adrConv,
 	}
 
-	irp.mutInterResultsForBlock.Lock()
 	irp.interResultsForBlock = make(map[string]*txInfo, 0)
-	irp.mutInterResultsForBlock.Unlock()
 
-	return nil, nil
+	return irp, nil
 }
 
 // CreateAllScrMiniBlocks returns the cross shard miniblocks for the current round created from the smart contract results
```

Comment:
Wrong comment. Please check all comments on exported funcs.

Reply:
done

Reply:
Forgot about the constructor. Also take a look at smartContractResults.go and goimports on transactions.go